pkg/promtail/client/client.go

```diff
@@ -191,11 +194,13 @@ func encodeBatch(batch map[model.Fingerprint]*logproto.Stream) ([]byte, error) {
 }
 
 func (c *Client) send(ctx context.Context, buf []byte) (int, error) {
+	timedCtx, cancel := context.WithTimeout(ctx, c.cfg.Timeout)
```

Comment:
This is a matter of opinion, but I'd overload the existing `ctx` here: 

```golang
ctx, cancel := context.WithTimeout(ctx, c.cfg.Timeout)
```

Its a pretty common pattern.